dialect/sql/sqlgraph/graph.go

```diff
@@ -1301,6 +1301,11 @@ func (c *creator) insertLastID(ctx context.Context, insert *sql.InsertBuilder) e
 		}
 		defer rows.Close()
 		if !c.ID.Type.Numeric() {
+			// If the ID implements the sql.Scanner
```

Comment:
Just out of curiosity: What about numeric types that have a custom `sql.Sanner` implementation?

Reply:
Good point. The `c.ID.Value.(field.ValueScanner)` check needs to move one level up. 

Thanks 🙏